dcos_test_utils/gce.py

```diff
@@ -0,0 +1,205 @@
+import logging
+
+from googleapiclient import discovery
+from googleapiclient.errors import HttpError
+from oauth2client.client import GoogleCredentials
+from retrying import retry
+
+from dcos_test_utils.helpers import Host
+
+log = logging.getLogger(__name__)
+
+
+class Instance:
+    def __init__(self, name, priv_ip, pub_ip):
+        self.name = name
+        self.private_ip = priv_ip
+        self.public_ip = pub_ip
+        self.host = Host(self.private_ip, self.public_ip)
+
+
+class Deployment:
+    def __init__(self, gce_wrapper, name, zone):
+        self.gce_wrapper = gce_wrapper
+        self.name = name
+        self.zone = zone
+        self.instances = []
+        self.fingerprint = None
+        self.errors = None
+
+    def get_instances(self):
```

Comment:
I like that you considered efficiency here by caching the results, but this unintentionally adds a lot of complexity we do not need.

I would recommend turning this into a generator property